costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -610,6 +609,16 @@ bool ObstacleLayer::adjustSensorOrigin(const Observation& clearing_observation,
   // find the intersection between the map and the line defined by the sensor ray
   bg::intersection(sensor_ray, map_boundary_, intersection_points);
 
+  // function to slightly shift the origin inwards to avoid floating point errors
+  const auto shift_inwards = [&ox, &oy, &wx, &wy]()
```

Comment:
super weak S: shift_inwards is a good name if u pass the origin as param.  As you do now, it should be shift_center_inwards